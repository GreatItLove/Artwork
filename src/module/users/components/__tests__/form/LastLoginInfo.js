import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import LastloginInfo from '../../form/LastloginInfo';

describe('LastloginInfo', () => {
  it('Snapshot of LastloginInfo without initialvalues', () => {
    const props = {
      initialValues: {}
    };
    const wrapper = shallow(<LastloginInfo {...props} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('Snapshot of LastloginInfo with initialvalues', () => {
    const props = {
      initialValues: {
        active: true,
        browserInfo: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_3) AppleWebKit/602.4.8 (KHTML, like Gecko) Version/10.0`,
        email: `ahershey79@yahoo.com`,
        fullName: `Andrew Hershey`,
        ip: `68.13.124.171`,
        lastLogin: `1501790500`,
        logins: ['1501790500', '1501790301'],
        numLogins: 3,
        userId: 6757,
        userLevel: 4,
        userType: `SA`
      }
    };
    const wrapper = shallow(<LastloginInfo {...props} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
