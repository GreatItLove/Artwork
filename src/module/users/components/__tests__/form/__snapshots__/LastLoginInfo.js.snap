// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastloginInfo Snapshot of LastloginInfo with initialvalues 1`] = `
<div
  className="py-4"
>
  <div
    className="col-sm-12"
  >
    <div
      className="row align-items-center"
    >
      <div
        className="col-md-4"
      >
        <label>
          View Logins:
        </label>
      </div>
      <div
        className="col-md-8"
      >
        <Field
          component={[Function]}
          label=""
          name="viewLogin"
          onChange={[Function]}
        />
      </div>
    </div>
  </div>
  <CSSTransition
    appear={true}
    classNames="fade"
    in={false}
    mountOnEnter={true}
    timeout={300}
    unmountOnExit={true}
  >
    <div
      className="mt-3 py-3 users lastLoginContainer"
    >
      <div
        className="col-sm-12 row align-items-center py-2"
      >
        <div
          className="col-md-4"
        >
          IP Address:
        </div>
        <div
          className="col-md-8"
        >
          68.13.124.171
        </div>
      </div>
      <div
        className="col-sm-12 row align-items-center py-2"
      >
        <div
          className="col-md-4"
        >
          Browser Info:
        </div>
        <div
          className="col-md-8"
        >
          Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_3) AppleWebKit/602.4.8 (KHTML, like Gecko) Version/10.0
        </div>
      </div>
      <div
        className="col-sm-12 row align-items-center py-2"
      >
        <div
          className="col-md-4"
        >
          Logins
        </div>
        <div
          className="col-md-8"
        >
          3
        </div>
      </div>
      <div
        className="col-sm-12 row align-items-center py-2"
      >
        <div
          className="col-md-4"
        >
          Last 5 Logins
        </div>
        <div
          className="col-sm-12 row align-items-center py-2"
        >
          <div
            className="col-md-4"
          />
          <div
            className="col-md-8 pl-4"
          >
            1. August 04, 2017 1:31 AM
          </div>
        </div>
        <div
          className="col-sm-12 row align-items-center py-2"
        >
          <div
            className="col-md-4"
          />
          <div
            className="col-md-8 pl-4"
          >
            2. August 04, 2017 1:28 AM
          </div>
        </div>
      </div>
    </div>
  </CSSTransition>
</div>
`;

exports[`LastloginInfo Snapshot of LastloginInfo without initialvalues 1`] = `
<div
  className="py-4"
>
  <div
    className="col-sm-12"
  >
    <div
      className="row align-items-center"
    >
      <div
        className="col-md-4"
      >
        <label>
          View Logins:
        </label>
      </div>
      <div
        className="col-md-8"
      >
        <Field
          component={[Function]}
          label=""
          name="viewLogin"
          onChange={[Function]}
        />
      </div>
    </div>
  </div>
  <CSSTransition
    appear={true}
    classNames="fade"
    in={false}
    mountOnEnter={true}
    timeout={300}
    unmountOnExit={true}
  >
    <div
      className="m-3"
    >
      No Login details found
    </div>
  </CSSTransition>
</div>
`;
